### PR TITLE
Adding JSNLog module [DDFLSBP-569]

### PR DESCRIPTION
**Pull request by clausbruun**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-569

#### Description

Add JSNLog module and corresponding ADR

#### Screenshot of the result

image

#### Sibling PR

https://github.com/danskernesdigitalebibliotek/dpl-react/pull/1257

### CODE REVIEW

**Review by Adamik10 — APPROVED** [review]
Very good! Love the ADR inclusion :))

**Review by Dresse — CHANGES_REQUESTED** [review]
LGTM 👍 

However after introducing config_ignore_auto, the core.extension.yml has been added to config_ignore. This means that new modules should be installed in an update hook in dpl_update module.

You can see it here:
https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/develop/web/modules/custom/dpl_update/dpl_update.install

**Comment by clausbruun** [comment]
> LGTM 👍
> 
> However after introducing config_ignore_auto, the core.extension.yml has been added to config_ignore. This means that new modules should be installed in an update hook in dpl_update module.
> 
> You can see it here: https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/develop/web/modules/custom/dpl_update/dpl_update.install

@Dresse: https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/1077/commits/1ec893443d56749feefc6a8095d67d9469fec909 :)

File: composer.json [hunk 1]
```diff
@@ -40,10 +40,10 @@
             "type": "package",
             "package": {
                 "name": "danskernesdigitalebibliotek/dpl-react",
-                "version": "2024.26.0",
+                "version": "2024.22.2-dev",
                 "type": "drupal-library",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/2024.26.0/dist-2024-26-0-4da2d1c24781dbec59896bd0152c7a7dc238d6ac.zip",
+                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/branch-feature%2FDDFLSBP-569-logging-i-frontend/dist-feature-ddflsbp-569-logging-i-frontend.zip",
```
Comment (spaceo) [inline composer.json:46]: Remember to remove this change

File: composer.json [hunk 2]
```diff
@@ -56,9 +56,9 @@
             "package": {
                 "name": "danskernesdigitalebibliotek/dpl-design-system",
                 "type": "drupal-library",
-                "version": "2024.25.0",
+                "version": "2024.22.1",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-design-system/releases/download/2024.25.0/dist-2024-25-0-44a88057ebe723f137f5ebe751cf0b6953c029b3.zip",
+                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-design-system/releases/download/branch-develop/dist-develop.zip",
```
Comment (spaceo) [inline composer.json:61]: Remember to remove this change

File: composer.json [hunk 3]
```diff
@@ -84,12 +84,13 @@
...
         "amazeeio/drupal_integrations": "^0.4",
+        "bower-asset/jsnlog": "^2.30",
         "brick/math": "^0.12.1",
         "composer/installers": "1.12.0",
         "cweagans/composer-patches": "1.7.3",
         "danskernesdigitalebibliotek/cms-api": "*",
-        "danskernesdigitalebibliotek/dpl-design-system": "^2024.25",
-        "danskernesdigitalebibliotek/dpl-react": "^2024.26",
+        "danskernesdigitalebibliotek/dpl-design-system": "^2024.22",
+        "danskernesdigitalebibliotek/dpl-react": "2024.22.2-dev",
```
Comment (spaceo) [inline composer.json:93]: Remember to remove these change

**Comment by Dresse** [comment]
> > LGTM 👍
> > However after introducing config_ignore_auto, the core.extension.yml has been added to config_ignore. This means that new modules should be installed in an update hook in dpl_update module.
> > You can see it here: https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/develop/web/modules/custom/dpl_update/dpl_update.install
> 
> @Dresse: [1ec8934](https://github.com/danskernesdigitalebibliotek/dpl-cms/commit/1ec893443d56749feefc6a8095d67d9469fec909) :)

Nice Claus!
I can see that the previous process has been to both enable modules in core.extensions.yml and in dpl_update. I was not clear about that earlier, I'm sorry!

Also I can see that it should also be added as part of the dpl_update_install hook in dpl_update.install

**Comment by clausbruun** [comment]
> Nice Claus! I can see that the previous process has been to both enable modules in core.extensions.yml and in dpl_update. I was not clear about that earlier, I'm sorry!
> 
> Also I can see that it should also be added as part of the dpl_update_install hook in dpl_update.install

@Dresse: Just to clarify, in your opinion, are there any further changes that I should make? :)

**Comment by Dresse** [comment]
> > Nice Claus! I can see that the previous process has been to both enable modules in core.extensions.yml and in dpl_update. I was not clear about that earlier, I'm sorry!
> > Also I can see that it should also be added as part of the dpl_update_install hook in dpl_update.install
> 
> @Dresse: Just to clarify, in your opinion, are there any further changes that I should make? :)

Yes, I think one more change is needed. 
We need to add this `$messages[] = dpl_update_update_10008();` to the dpl_update_install hook in dpl_update.install file 👍

**Review by Dresse — APPROVED** [review]
LGTM 👍

**Comment by clausbruun** [comment]
> Yes, I think one more change is needed. We need to add this `$messages[] = dpl_update_update_10008();` to the dpl_update_install hook in dpl_update.install file 👍

Done: https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/1077/commits/b33791c3cb312cad9607ae20095af912b59c92a0

**Pull request merged**